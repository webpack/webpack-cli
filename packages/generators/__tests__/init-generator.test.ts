import { join } from 'path';
import { run } from 'yeoman-test';

// fixme: unstable
describe.skip('init generator', () => {
    it('generates a webpack project config', async () => {
        const outputDir = await run(join(__dirname, '../init-generator')).withPrompts({
            multiEntries: false,
            singularEntry: 'src/index2',
            outputDir: 'dist2',
            langType: 'No',
            stylingType: 'No',
            useExtractPlugin: 'main',
        });
        const output = require(join(outputDir, '.yo-rc.json'));
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        const config = (Object.entries(output)[0][1] as any).configuration.config.webpackOptions;
        expect(config.entry).toEqual("'./src/index2.js'");
        expect(config.output.path).toEqual("path.resolve(__dirname, 'dist2')");
    });
});
