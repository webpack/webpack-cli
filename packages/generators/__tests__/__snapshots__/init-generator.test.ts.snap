// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`init generator generates a webpack config that uses ES6 1`] = `
Object {
  "devServer": Object {
    "host": "'localhost'",
    "open": true,
  },
  "mode": "'development'",
  "module": Object {
    "rules": Array [
      Object {
        "include": Array [
          "path.resolve(__dirname, 'src')",
        ],
        "loader": "'babel-loader'",
        "test": "/\\\\.(js|jsx)$/",
      },
    ],
  },
  "plugins": Array [
    "new webpack.ProgressPlugin()",
    "new WorkboxWebpackPlugin.GenerateSW({
				swDest: 'sw.js',
				clientsClaim: true,
				skipWaiting: false,
			})",
  ],
}
`;

exports[`init generator generates a webpack config that uses Typescript 1`] = `
Object {
  "devServer": Object {
    "host": "'localhost'",
    "open": true,
  },
  "entry": "'./src/index.ts'",
  "mode": "'development'",
  "module": Object {
    "rules": Array [
      Object {
        "exclude": Array [
          "/node_modules/",
        ],
        "include": Array [
          "path.resolve(__dirname, 'src')",
        ],
        "loader": "'ts-loader'",
        "test": "/\\\\.(ts|tsx)$/",
      },
    ],
  },
  "plugins": Array [
    "new webpack.ProgressPlugin()",
    "new WorkboxWebpackPlugin.GenerateSW({
				swDest: 'sw.js',
				clientsClaim: true,
				skipWaiting: false,
			})",
  ],
  "resolve": Object {
    "extensions": Array [
      "'.tsx'",
      "'.ts'",
      "'.js'",
    ],
  },
}
`;

exports[`init generator generates a webpack config that uses Typescript 2`] = `
Object {
  "compilerOptions": Object {
    "allowJs": true,
    "allowSyntheticDefaultImports": true,
    "module": "es6",
    "noImplicitAny": true,
    "target": "es5",
  },
  "files": Array [
    "src/index.ts",
  ],
}
`;

exports[`init generator generates a webpack config using CSS with mini-css-extract-plugin 1`] = `
Object {
  "devServer": Object {
    "host": "'localhost'",
    "open": true,
  },
  "mode": "'development'",
  "module": Object {
    "rules": Array [
      Object {
        "test": "/.css$/",
        "use": Array [
          Object {
            "loader": "MiniCssExtractPlugin.loader",
          },
          Object {
            "loader": "\\"css-loader\\"",
            "options": Object {
              "sourceMap": true,
            },
          },
        ],
      },
    ],
  },
  "plugins": Array [
    "new webpack.ProgressPlugin()",
    "new MiniCssExtractPlugin({ filename:'main.[contenthash].css' })",
    "new WorkboxWebpackPlugin.GenerateSW({
				swDest: 'sw.js',
				clientsClaim: true,
				skipWaiting: false,
			})",
  ],
}
`;

exports[`init generator generates a webpack config using CSS without mini-css-extract-plugin 1`] = `
Object {
  "devServer": Object {
    "host": "'localhost'",
    "open": true,
  },
  "mode": "'development'",
  "module": Object {
    "rules": Array [
      Object {
        "test": "/.css$/",
        "use": Array [
          Object {
            "loader": "\\"style-loader\\"",
          },
          Object {
            "loader": "\\"css-loader\\"",
            "options": Object {
              "sourceMap": true,
            },
          },
        ],
      },
    ],
  },
  "plugins": Array [
    "new webpack.ProgressPlugin()",
    "new WorkboxWebpackPlugin.GenerateSW({
				swDest: 'sw.js',
				clientsClaim: true,
				skipWaiting: false,
			})",
  ],
}
`;

exports[`init generator generates a webpack config with custom entry and output 1`] = `
Object {
  "devServer": Object {
    "host": "'localhost'",
    "open": true,
  },
  "entry": "'./src/index2.js'",
  "mode": "'development'",
  "module": Object {
    "rules": Array [],
  },
  "output": Object {
    "path": "path.resolve(__dirname, 'dist2')",
  },
  "plugins": Array [
    "new webpack.ProgressPlugin()",
    "new WorkboxWebpackPlugin.GenerateSW({
				swDest: 'sw.js',
				clientsClaim: true,
				skipWaiting: false,
			})",
  ],
}
`;

exports[`init generator generates a webpack config with default options 1`] = `
Object {
  "devServer": Object {
    "host": "'localhost'",
    "open": true,
  },
  "mode": "'development'",
  "module": Object {
    "rules": Array [],
  },
  "plugins": Array [
    "new webpack.ProgressPlugin()",
    "new WorkboxWebpackPlugin.GenerateSW({
				swDest: 'sw.js',
				clientsClaim: true,
				skipWaiting: false,
			})",
  ],
}
`;

exports[`init generator generates a webpack config with multiple entries 1`] = `
Object {
  "devServer": Object {
    "host": "'localhost'",
    "open": true,
  },
  "entry": Object {
    "test1": "'./dir1/test1.js'",
    "test2": "'./dir2/test2.js'",
  },
  "mode": "'development'",
  "module": Object {
    "rules": Array [],
  },
  "plugins": Array [
    "new webpack.ProgressPlugin()",
    "new WorkboxWebpackPlugin.GenerateSW({
				swDest: 'sw.js',
				clientsClaim: true,
				skipWaiting: false,
			})",
  ],
}
`;

exports[`init generator generates a webpack config with optional dependencies 1`] = `
Object {
  "devServer": Object {
    "host": "'localhost'",
    "open": true,
  },
  "mode": "'development'",
  "module": Object {
    "rules": Array [],
  },
  "plugins": Array [
    "new webpack.ProgressPlugin()",
    "new HtmlWebpackPlugin({
					template: 'index.html'
				})",
    "new WorkboxWebpackPlugin.GenerateSW({
				swDest: 'sw.js',
				clientsClaim: true,
				skipWaiting: false,
			})",
  ],
}
`;
